### ISSUE TITLE
[tracking issue] Jaeger Clients should not allow null or empty service names

### ISSUE THREAD

**Issue by vprithvi** [jaegertracing/jaeger#1107]
## Requirement - what kind of business use case are you trying to solve?
Ensure that the service name field is filled when Jaeger clients are created. 

## Problem - what in Jaeger blocks you from solving the requirement?
- This would be a breaking change when we start enforcing this requirement

## Proposal - what do you suggest to solve the problem or improve the existing situation?
- If the service name is null, or empty - prevent the Tracer object from being created.  Implementations might differ across languages. 

## Any open questions to address
- This is an instance of validating input parameters. Do other parameters need validation?

### TODO: Create issues in all client libraries

**Comment by jpkrohling** [comment]
This is true already for the Java client, for instance. If other libraries aren't enforcing it, I think they should, at least for consistency's sake...

https://github.com/jaegertracing/jaeger-client-java/blob/6ee913a2d88c34f311d46b0b3e8928514525f62a/jaeger-core/src/test/java/io/jaegertracing/internal/JaegerTracerTest.java#L106-L114
